data/shared/citizen/scripting/v8/main.js

```diff
@@ -90,14 +90,27 @@ const EXT_LOCALFUNCREF = 11;
 					case 0:
 						return undefined;
 					case 1:
-						return retvals[0];
+						const rv = retvals[0];
+						if (rv && rv['__cfx_async_retval']) {
+							return new Promise((res, rej) => {
+								rv['__cfx_async_retval']((v, e) => {
+									if (e != null)
+										rej(e);
+									else
+										res(v);
```

Comment:
This doesn't seem to transparently unwrap array values like above (1 = return directly, 0 = undefined, 2 = array).

Reply:
Should be resolved in next commit. Also I´ve encountered issue while catching errors from __cfx_async_retval. After serializing the error object is sent as:
![image](https://user-images.githubusercontent.com/19434539/132693273-85b3f8d6-6e7e-40de-9359-83ef1d3b0d9d.png)

Reply:
Actually I could get error message by returning `err.toString()` or `err.message` inside `setCallRefFunction` but unhandled promise rejection won´t throw error this way.